src/OpenTelemetry.AutoInstrumentation/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,79 @@
+// <copyright file="PluginManager.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.AutoInstrumentation.Configuration;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.AutoInstrumentation.Plugins;
+
+internal class PluginManager
+{
+    private readonly IReadOnlyList<(Type Type, object Instance)> _plugins;
+
+    public PluginManager(GeneralSettings settings)
+    {
+        var plugins = new List<(Type, object)>();
+
+        foreach (var assemblyQualifiedName in settings.Plugins)
+        {
+            var type = Type.GetType(assemblyQualifiedName, throwOnError: true);
+            var instance = Activator.CreateInstance(type);
+
+            plugins.Add((type, instance));
+        }
```

Comment:
May be worth adding a comment that would explain why we are doing this in the constructor

As far as I understand it gives us the benefit that we will "fail-fast" in case of wrong plugin configuration.

The downside is that we have a memory footprint as the `_plugins` field is kept for the whole application lifecycle.

Probably we could consider just "validate" the configuration using reflection in the constructor (to have the "fail-fast") and call `Activator.CreateInstance` on demand in the `Configure` methods.

Reply:
we could consider also "disposing" the plugins if they implement `IDisposable`